pkg/snapshot/snapshot.go

```diff
@@ -0,0 +1,99 @@
+package snapshot
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hyperledger/fabric-admin-sdk/internal/protoutil"
+	"github.com/hyperledger/fabric-admin-sdk/pkg/identity"
+	"github.com/hyperledger/fabric-protos-go-apiv2/common"
+	"github.com/hyperledger/fabric-protos-go-apiv2/peer"
+	"google.golang.org/grpc"
+	"google.golang.org/protobuf/proto"
+)
+
+// Client is a wrapper around snapshot client
+type Client struct {
+	snapshot peer.SnapshotClient
+	id       identity.SigningIdentity
+}
+
+func New(conn grpc.ClientConnInterface, id identity.SigningIdentity) *Client {
+	return &Client{
+		snapshot: peer.NewSnapshotClient(conn),
+		id:       id,
+	}
+}
+
+// SubmitSnapshotRequest from a specific peer
+func (s *Client) SubmitSnapshotRequest(ctx context.Context, channelID string, blockNum uint64) error {
+	signedRequest, err := s.newSignedSnapshotRequest(channelID, blockNum)
+	if err != nil {
+		return fmt.Errorf("failed to create signed snapshot request: %w", err)
+	}
+
+	if _, err := s.snapshot.Generate(ctx, signedRequest); err != nil {
+		return fmt.Errorf("failed to submit generate snapshot request: %w", err)
+	}
+
+	return nil
+}
+
+// CancelSnapshotRequest from a specific peer
+func (s *Client) CancelSnapshotRequest(ctx context.Context, channelID string) error {
```

Comment:
Perhaps this can be called just `CancelRequest` or even `Cancel`? It should probably mirror the naming of the `SubmitRequest` method. If that is called `SubmitRequest`, this makes sense as `CancelRequest`. If the submit is called `Generate` (or `Submit`) then this makes more sense as the shorter `Cancel`.